src/http/common.cr

```diff
@@ -139,6 +139,13 @@ module HTTP
   end
 
   def self.serialize_headers_and_string_body(io, headers, body)
+    if headers.includes_word?("Content-Encoding", "gzip")
+      body = IO::Memory.new.tap do |buf|
+        gzip = Gzip::Writer.new(buf)
+        gzip.print body
```

Comment:
I think this is a bit confusing with the two bodies and the tap, could you clean this up?

Reply:
Yup. Finally I got two candidates about this.
```crystal
      body = IO::Memory.new.tap do |buf|
        Gzip::Writer.open buf, &.print body
      end
```

and

```crystal
      buf = IO::Memory.new
      Gzip::Writer.open buf, &.print body
      body = buf
```

I like the latter. Because although the code is a bit redundant, it's easy to read due to no nested bodies. Right?

Reply:
I would format it a bit nicer though, the whitespace is very confusing here.#

```cr
      buf = IO::Memory.new
      Gzip::Writer.open(buf, &.print(body))
      body = buf
```